geoai/agents/geo_agents.py

```
@@ -287,6 +287,46 @@ def create_minimax_model(
     return OpenAIModel(client_args=client_args, model_id=model_id, **kwargs)
 
 
+def create_vllm_model(
+    base_url: str = "http://localhost:8000/v1",
+    model_id: str = "meta-llama/Llama-3.1-8B-Instruct",
+    api_key: str = "EMPTY",
+    client_args: dict = None,
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)